pkg/plugins/mellanox/mellanox_plugin.go

```diff
@@ -176,46 +172,37 @@ func runCommand(command string, args ...string) (string, error) {
 }
 
 func getMlnxNicFwData(pciAddress string) (current, next *mlnxNic, err error) {
-	glog.Infof("mellanox-plugin getMlnxNicFwData(): for device %s", pciAddress)
+	glog.Infof("mellanox-plugin getMlnxNicFwData(): device %s", pciAddress)
 	err = nil
 	attrs := []string{TotalVfs, EnableSriov}
-	singlePort, err := isSinglePortNic(pciAddress)
-	if err != nil {
-		glog.Errorf("mellanox-plugin getMlnxNicFwData(): %v", err)
-		return
-	}
 	firstPort := isFirstPort(pciAddress)
 
-	if singlePort {
-		attrs = append(attrs, "LINK_TYPE")
+	if firstPort {
+		attrs = append(attrs, "LINK_TYPE_P1")
 	} else {
-		if firstPort {
-			attrs = append(attrs, "LINK_TYPE_P1")
-		} else {
-			attrs = append(attrs, "LINK_TYPE_P2")
-		}
+		attrs = append(attrs, "LINK_TYPE_P2")
 	}
-	out, err := mstconfigReadData(pciAddress, attrs...)
+	out, err := mstConfigReadData(pciAddress)
 	if err != nil {
 		return
 		glog.Errorf("mellanox-plugin getMlnxNicFwData(): %v", err)
```

Comment:
dead code, move above return, also the content should express that we failed to query mstconfig

Reply:
fixed